### PR TITLE
refactor(makeText): Rename variables, move variables to local scope

### PR DESCRIPTION
**Pull request by bryanjenningz**
The goal of this pull request is to give variables in `makeText` more descriptive names so that the code is easier to understand. Some variables were also put into a more local scope so that they're easier to follow.

This pull request does not change the functionality of `makeText` - the copy-to-clipboard feature works exactly how it did before this pull request.

I tried to make this pull request have a small code diff so that it's easier to review. I plan on making more changes to `makeText` in subsequent pull requests which will make it even easier to understand.

Changes I plan on making in subsequent pull requests:
- Restructure the `kanjiInfoLabelList` array so that it's an array of objects - so its type might change to something like `{ kanjiInfoCode: string; kanjiInfoName: string }[]` so that it's easier to understand.
- Write a dict entry parsing function which is well-tested and unifies the parsing logic used in `makeText`, `makeHtml`, and the update-db.ts file.
- Change the `DictEntryData` type to represent the valid states of the application better - something like `type DictEntryData = KanjiDictEntry | WordDictEntry;`

### CODE REVIEW

File: extension/data.ts
Comment (melink14) [inline extension/data.ts:947]: In the long run, this type of structured output should probably be done with a template (such that the final text is easier to see in the code).

For HTML, I would naturally assume `lit-html` would be a good candidate since I use it for the options page.

However, I'm not sure if it would work for generating non-html text (haven't tried). If not maybe something like normal template literals would improve things (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) but we'd need to do something special for the list inputs...

Not a request to change anything but something I thought of when seeing your improvements!

File: extension/data.ts
Comment (melink14) [inline extension/data.ts:968]: Thinking about these hyphens. I wonder if it would be better to replace empty string with hyphens from the very beginning (when we read the data from the dictionary) so we don't need to check so often later.

This one could at least be moved to line 966 but that depends on how things look after later refactors.

File: extension/data.ts
Comment (melink14) [inline extension/data.ts:977]: this is ripe to be converted to named capture groups probably. Just an FYI of course!

File: extension/data.ts
Comment (melink14) [inline extension/data.ts:979]: The codecov failure is because this branch isn't covered I believe. (the first branch at the top is also not covered but since you didn't change those lines you shouldn't be impacted).

Since that's just a limitation of the coverage algorithm, there's no need to solve it now if you don't want. But I'll need to override the check.

**Review by melink14 — APPROVED** [review]
Great work keeping the PR small. It looks good to me with just some FYI comments. After you read/resolve stuff I'll merge it directly!

I hope you're not blocked by the review process since you can keep working on top of these changes with stacked branches!

File: extension/data.ts
Comment (bryanjenningz) [reply extension/data.ts:978]: That's a good idea - it would make the regex more readable. We can do this in a later pull request to keep this pull request small.

File: extension/data.ts
Comment (bryanjenningz) [reply extension/data.ts:980]: I decided to add some tests to get full test coverage for `makeText` so this problem is resolved now.

File: extension/data.ts
Comment (bryanjenningz) [reply extension/data.ts]: This is a good idea. I decided to do this with both `kanjiInfo` and `kanjiInfoName`.

File: extension/data.ts
Comment (bryanjenningz) [reply extension/data.ts:947]: Yeah, that might make the code feel more declarative and easier to follow. I'll keep that in mind in the future.

File: extension/test/data_test.ts
Comment (bryanjenningz) [inline extension/test/data_test.ts:144]: We might restructure the data types for the dict entries in the future so that the values are parsed and checked before this function is even called, so we might be able to delete these tests in the future.

I'll leave these tests in for now just so the current behavior is documented and so we won't have to deal with code coverage problems in `makeText` for now.

File: extension/test/data_test.ts
Comment (melink14) [inline extension/test/data_test.ts:90]: (super nit) It's a bit strange to call this `emptyString` already since that's what we're asserting later. Usually, this would just be something like `result` or `text`. By embedding the expecation in the actual value it might lead readers to not read the assertion line as closely leading to eventual drift.

I actually realize now this is similar to what you've done in the other tests and I wouldn't change it in this PR.

**Review by melink14 — APPROVED** [review]

File: extension/test/data_test.ts
Comment (bryanjenningz) [reply extension/test/data_test.ts:90]: Yeah, that makes sense. I can change all the variable names so they aren't like this.

**Pull request merged**
